File: 2015/01/floors.py
def part_one():
    with open("floors.txt") as file:
        lines = list(file.readlines())
    floor = 0
    for line in lines:
        for char in line:
            if char == "(":
                floor += 1
            elif char == ")":
                floor -= 1 
    return floor

def part_two():
    with open("floors.txt") as file:
        lines = list(file.readlines())
    floor = 0
    for line in lines:
        for index, char in enumerate(line):
            if char == "(":
                floor += 1
            elif char == ")":
                floor -= 1 
            if floor == -1:
                return index + 1

if __name__ == "__main__":
    print(f"Part One: {part_one()}")
    print(f"Part Two: {part_two()}")

File: 2015/01/Floors.java
import java.util.*;
import java.io.*;

public class Floors {
  public static void main(String[] args) throws FileNotFoundException {
    Scanner readIn = new Scanner(new File("floors.txt"));
    String line = readIn.nextLine();
    int floor = 0;
    int first_neg = -1;
    for (int i = 0; i < line.length(); i++) {
      if (line.substring(i, i+1).equals("(")) {
        floor++;
      } else if (line.substring(i, i+1).equals(")")) {
        floor--;
        if (floor < 0 && first_neg == -1) {
          first_neg = i + 1;
        }
      }
    }

    System.out.println("Part One: " + floor);
    System.out.println("Part Two: " + first_neg);
  }
}
